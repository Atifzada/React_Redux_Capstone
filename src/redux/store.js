import { configureStore, combineReducers } from '@reduxjs/toolkit';
import cryptoReducer from './cryptoSlice';
import detailsReducer from './detailSlice';

const reducer = combineReducers({
  crypto: cryptoReducer,
  cryptoDetails: detailsReducer,
});

const store = configureStore({ reducer });

export default store;
