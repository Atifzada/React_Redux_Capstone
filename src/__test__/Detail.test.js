import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import CryptoDetails from '../routes/Detail';

describe('CryptoDetail page testing', () => {
  test('renders correctly when crypto details are available', () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <CryptoDetails />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
