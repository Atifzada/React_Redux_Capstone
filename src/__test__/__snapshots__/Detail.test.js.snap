// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoDetail Component renders correctly when crypto details are available 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="detailsCard myflex"
      >
        <h2
          class="detailsTitle"
        />
        <img
          alt="crypto icon"
          class="detailsIcon"
        />
        <div
          class="detailsContainer"
        >
          <div
            class="detailsSpecifications detailsBG"
          >
            <span
              class="detailsText"
            >
              Abbrivation:
            </span>
          </div>
          <div
            class="detailsSpecifications"
          >
            <span
              class="detailsText"
            >
              MarketCap: $
            </span>
          </div>
          <div
            class="detailsSpecifications detailsBG"
          >
            <span
              class="detailsText"
            >
              Changes Rate (1H) :
            </span>
          </div>
          <div
            class="detailsSpecifications"
          >
            <a />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="detailsCard myflex"
    >
      <h2
        class="detailsTitle"
      />
      <img
        alt="crypto icon"
        class="detailsIcon"
      />
      <div
        class="detailsContainer"
      >
        <div
          class="detailsSpecifications detailsBG"
        >
          <span
            class="detailsText"
          >
            Abbrivation:
          </span>
        </div>
        <div
          class="detailsSpecifications"
        >
          <span
            class="detailsText"
          >
            MarketCap: $
          </span>
        </div>
        <div
          class="detailsSpecifications detailsBG"
        >
          <span
            class="detailsText"
          >
            Changes Rate (1H) :
          </span>
        </div>
        <div
          class="detailsSpecifications"
        >
          <a />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
